### PR TITLE
Create test (#250)

### PR DESCRIPTION
**Pull request by hani-iterable**
## JIRA Ticket

* [MOB-7534](https://iterable.atlassian.net/browse/MOB-7534)

## Description

- **Release Notes**
> The required changes for correct email/userId field placement have been completed

> We've also updated the sample app to showcase the changes.

- **Test Steps**
> Start by launching the sample app, then select the 'embedded' option. Enter your userId/email in the designated field. Afterward, choose the option you wish to test and click submit button. You will then be able to observe the parameter changes in the API request.

[MOB-7534]: https://iterable.atlassian.net/browse/MOB-7534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: react-example/src/components/EmbeddedForm.tsx [hunk 1]
```diff
@@ -102,12 +103,13 @@ export const EmbeddedForm: FC<Props> = ({
     const appPackageName = 'my-lil-site';
 
     trackEmbeddedMessageClick(
-      userId,
       payload,
       buttonIdentifier,
       clickedUrl,
       appPackageName,
-      1627060811283
+      1627060811283,
```
Comment (mprew97) [inline react-example/src/components/EmbeddedForm.tsx:110]: should this actually be hardcoded?

File: react-example/src/components/EmbeddedForm.tsx [hunk 2]
```diff
@@ -154,7 +156,8 @@ export const EmbeddedForm: FC<Props> = ({
     setTrackingEvent(true);
 
     const sessionData = {
-      userId: userId,
+      userId: emailRegex.test(userId) ? undefined : userId,
```
Comment (mprew97) [inline react-example/src/components/EmbeddedForm.tsx:159]: you should be able to just pass in the value but determine which key to use dynamically. would also suggest making a helper/util method to do so.

Comment (mprew97) [reply react-example/src/components/EmbeddedForm.tsx:159]: See https://www.freecodecamp.org/news/how-to-set-dynamic-object-properties-using-computed-property-names/

**Review by mprew97 — CHANGES_REQUESTED** [review]

File: react-example/src/components/EmbeddedForm.tsx (hunk 1 above)
Comment (hani-iterable) [reply react-example/src/components/EmbeddedForm.tsx]: I have changed to actual value.

File: react-example/src/components/EmbeddedForm.tsx (hunk 2 above)
Comment (hani-iterable) [reply react-example/src/components/EmbeddedForm.tsx]: Modified both the SDK and the sample app to incorporate support for 'Computed Property Names' wherever applicable.

**Review by mprew97 — APPROVED** [review]

**Pull request merged**